frontend/src/components/Search/Search.jsx

```diff
@@ -64,12 +65,13 @@ class Search extends React.Component {
 
     onChange = (e) => {
         this.props.setParams({qTemp: e.target.value});
-        this.search(e.target.value);
     }
 
-    search = debounce((q) => {
-        this.props.setParams({q});
-    }, 500)
+    onKeyDown = (e) => {
```

Comment:
just a thought: I wonder if it'd be worth it to add a "Search" button so it's a bit more clear that searching now requires an action (pressing Enter or clicking the button)